### PR TITLE
Chore: deprecate refreshClients methods

### PR DESCRIPTION
**Pull request by felipe-rod123**
## Proposed changes (including videos or screenshots)

Created the 'assets.refreshClients' endpoint for the `apps/meteor/client/views/admin/settings/groups/AssetsGroupPage.tsx` file.

### CODE REVIEW

**Review by ggazzo — APPROVED** [review]

**Review by sampaiodiego — CHANGES_REQUESTED** [review]
@ggazzo should we remove this functionality completely? the button "apply and refresh clients" doesn't work at all.. no reason to keep it there.

File: packages/rest-typings/src/v1/assets.ts [hunk 1]
```diff
@@ -14,6 +14,11 @@ export type AssetsEndpoints = {
 	'/v1/assets.unsetAsset': {
 		POST: (params: AssetsUnsetAssetProps) => void;
 	};
+
+	/** @deprecated */
+	'/v1/assets.refreshClients': {
+		POST: () => void;
+	};
```
Comment (ggazzo) [inline packages/rest-typings/src/v1/assets.ts:21]:
```suggestion
```

File: apps/meteor/tests/end-to-end/api/14-assets.js [hunk 2]
```diff
@@ -82,4 +82,29 @@ describe('[Assets]', function () {
...
+				.post(api('assets.refreshClients'))
+				.set({})
+				.expect('Content-Type', 'application/json')
+				.expect(401)
+				.expect((res) => {
+					expect(res.body).to.have.property('message');
+				})
+				.end(done);
+		});
+	});
```
Comment (ggazzo) [inline apps/meteor/tests/end-to-end/api/14-assets.js:109]:
```suggestion
```

File: apps/meteor/client/views/admin/settings/groups/AssetsGroupPage.tsx [hunk 3]
```diff
@@ -12,33 +10,9 @@ type AssetsGroupPageProps = ISetting;
...
-		<GroupPage
-			_id={_id}
-			{...group}
-			headerButtons={
-				<>
-					<Button onClick={handleApplyAndRefreshAllClientsButtonClick}>{t('Apply_and_refresh_all_clients')}</Button>
-				</>
-			}
-		>
+		<GroupPage _id={_id} {...group} headerButtons={<></>}>
```
Comment (ggazzo) [inline apps/meteor/client/views/admin/settings/groups/AssetsGroupPage.tsx:15]:
```suggestion
		<GroupPage _id={_id} {...group}>
```

File: apps/meteor/app/api/server/v1/assets.ts [hunk 4]
```diff
@@ -61,3 +62,17 @@ API.v1.addRoute(
...
+		authRequired: true,
+	},
+	{
+		post() {
+			methodDeprecationLogger.warn('assets.refreshClients will be deprecated in future versions of Rocket.Chat');
+			Meteor.call('refreshClients');
+			return API.v1.success();
+		},
+	},
+);
```
Comment (ggazzo) [inline apps/meteor/app/api/server/v1/assets.ts:78]:
```suggestion
```

File: apps/meteor/app/api/server/v1/assets.ts [hunk 5]
```diff
@@ -4,6 +4,7 @@ import { isAssetsUnsetAssetProps } from '@rocket.chat/rest-typings';
 import { RocketChatAssets } from '../../../assets/server';
 import { API } from '../api';
 import { getUploadFormData } from '../lib/getUploadFormData';
+import { methodDeprecationLogger } from '../../../lib/server/lib/deprecationWarningLogger';
```
Comment (ggazzo) [inline apps/meteor/app/api/server/v1/assets.ts:7]:
```suggestion
```

**Review by ggazzo — APPROVED** [review]

**Pull request merged**